homeassistant/components/essent/sensor.py

```diff
@@ -0,0 +1,280 @@
+                "end": _format_dt_str(tariff.end),
+            }
+    return {}
+
+
+SENSORS: tuple[EssentSensorEntityDescription, ...] = (
+    EssentSensorEntityDescription(
+        key="current_price",
+        name="current price",
+        translation_key="current_price",
+        value_fn=lambda entity: (
+            None
+            if (tariff := _get_current_tariff(entity)) is None
+            else tariff.total_amount
+        ),
+        attrs_fn=_current_price_attrs,
+    ),
+    EssentSensorEntityDescription(
+        key="next_price",
+        name="next price",
+        translation_key="next_price",
+        value_fn=lambda entity: (
+            None
+            if (tariff := _get_next_tariff(entity)) is None
+            else tariff.total_amount
+        ),
+        attrs_fn=_next_price_attrs,
+    ),
+    EssentSensorEntityDescription(
+        key="average_today",
+        name="average today",
+        translation_key="average_today",
+        value_fn=lambda entity: None
+        if (data := entity.energy_data) is None
+        else data.avg_price,
+        attrs_fn=_average_attrs,
+    ),
+    EssentSensorEntityDescription(
+        key="lowest_price_today",
+        name="lowest price today",
+        translation_key="lowest_price_today",
+        value_fn=lambda entity: None
+        if (data := entity.energy_data) is None
+        else data.min_price,
+        attrs_fn=lambda entity: (
+            {}
+            if (data := entity.energy_data) is None
+            else _extreme_attrs(entity, data.min_price)
+        ),
+        energy_types=(ENERGY_TYPE_ELECTRICITY,),
+        entity_registry_enabled_default=False,
+    ),
+    EssentSensorEntityDescription(
+        key="highest_price_today",
+        name="highest price today",
+        translation_key="highest_price_today",
+        value_fn=lambda entity: None
+        if (data := entity.energy_data) is None
+        else data.max_price,
+        attrs_fn=lambda entity: (
+            {}
+            if (data := entity.energy_data) is None
+            else _extreme_attrs(entity, data.max_price)
+        ),
+        energy_types=(ENERGY_TYPE_ELECTRICITY,),
+        entity_registry_enabled_default=False,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: EssentConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up Essent sensors."""
+    coordinator: EssentDataUpdateCoordinator = entry.runtime_data
+    entities: list[EssentSensor] = []
+
+    for description in SENSORS:
+        for energy_type in description.energy_types:
+            entities.append(EssentSensor(coordinator, energy_type, description))
+
+    async_add_entities(entities)
+
+
+class EssentSensor(EssentEntity, SensorEntity):
+    """Generic Essent sensor driven by entity descriptions."""
+
+    _attr_device_class = SensorDeviceClass.MONETARY
+    _attr_state_class = None
```

Comment:
Monetary device class should only be used for balances, not for the cost of things IIRC

Reply:
I looked at the Nord Pool integration and this doesn't specify a device class at all. So i copied that approach and removed it here as well.